lang/py/avro/schema.py

```diff
@@ -963,6 +1044,25 @@ def to_json(self, names=None):
         to_dump['fields'] = [f.to_json(names) for f in self.fields]
         return to_dump
 
+    def to_canonical_json(self, names):
+        if names is None:
+            names = Names()
+
+        if self.type == 'request':
+            raise NotImplementedError("Canonical form (probably) does not make sense on type request")
+
+        to_dump = self.canonical_properties
+        to_dump["name"] = self.fullname
+
+        if self.fullname in names.names:
+            return self.name_ref(names)
+        else:
```

Comment:
No need for else after return

Reply:
👍 Addressed in eacfdd6a0cd6de9b4e15701d08d9e994f09be310